include/flatbuffers/idl.h

```diff
@@ -59,6 +59,21 @@ namespace flatbuffers {
   TD(STRUCT, "",       Offset<void>, int, int, int, int) \
   TD(UNION,  "",       Offset<void>, int, int, int, int)
 
+#define FLATBUFFERS_GEN_TYPES_SCALAR_ALIAS(TD) \
```

Comment:
Sorry, I misread you comment I though you meant GEN_TYPES_ALIAS -> GEN_TYPES_ALIAS_SCALAR.

If I used GEN_TYPES I would affect all current and future backends. I preferred to not touch all backend code generators, and I think it would be better to separate out the macro to one per backend.

The other issue was originally that I needed a separate token type, but that could be shoehorned into the above - but it would not be clean since the other types are backend types and only the first (ENUM) is a token.